java/src/com/google/idea/blaze/java/sync/projectstructure/Jdks.java

```diff
@@ -122,16 +121,23 @@ private static boolean jdkPathMatches(Sdk jdk, String homePath) {
   @VisibleForTesting
   static Sdk findClosestMatch(LanguageLevel langLevel) {
     return ProjectJdkTable.getInstance().getSdksOfType(JavaSdk.getInstance()).stream()
-        .filter(
-            sdk -> {
-              LanguageLevel level = getJavaLanguageLevel(sdk);
-              return level != null && level.isAtLeast(langLevel);
-            })
+        .filter(sdk -> isLanguageLevelSupportedBySdkLevel(langLevel, getJavaLanguageLevel(sdk)))
         .filter(Jdks::isValid)
         .min(Comparator.comparing(Jdks::getJavaLanguageLevel))
         .orElse(null);
   }
 
+  private static boolean isLanguageLevelSupportedBySdkLevel(LanguageLevel requestedLanguageLevel,
+      LanguageLevel sdkLanguageLevel) {
+    if (sdkLanguageLevel == null) {
+      return false;
+    } else if (requestedLanguageLevel.isPreview()) {
+      return requestedLanguageLevel.toJavaVersion().equals(sdkLanguageLevel.toJavaVersion());
```

Comment:
you right; that was my misunderstanding. I've updated some JDK selection methods to check if a JDK supports a given language level instead of assuming that a JDK has only a single language level.